packages/webpack-plugin/src/runtime.ts

```diff
@@ -0,0 +1,142 @@
+        }
+
+        var valueAsString = stateValue.toString();
+
+        return createStateWithParamClassName(namespace, stateName, valueAsString);
+    }
+
+    function style(namespace: string) {
+        var classNames = [];
+
+        for (var i = 1; i < arguments.length; i++) {
+            // eslint-disable-next-line prefer-rest-params
+            var item = arguments[i];
+
+            if (item) {
+                if (typeof item === 'string') {
+                    classNames[classNames.length] = item;
+                } else if (i === 2) {
+                    for (var stateName in item) {
+                        var stateValue = item[stateName];
+                        var stateClass = createStateClass(namespace, stateName, stateValue);
+                        if (stateClass) {
+                            classNames[classNames.length] = stateClass;
+                        }
+                    }
+                }
+            }
+        }
+        return classNames.join(' ');
+    }
+
+    function cssStates(namespace: string, stateMapping?: StateMap | null): string {
+        var classNames = [];
+        for (var stateName in stateMapping) {
+            var stateValue = stateMapping[stateName];
+            var stateClass = createStateClass(namespace, stateName, stateValue);
+            if (stateClass) {
+                classNames.push(stateClass);
+            }
+        }
+        return classNames.join(' ');
+    }
+
+    host.sts = style;
+    host.stc = cssStates;
+}
+
+export function injectStyles(host: Host) {
+    function stylableRuntime(
+        namespace: string,
+        css: string,
+        depth: number,
+        runtimeId: string
+    ): void {
+        if (typeof document === 'undefined') {
+            return;
+        }
+        var d = document;
+        var head = d.head;
+        var style = d.createElement('style');
+        style.setAttribute('st-depth', (depth as unknown) as string);
+        style.setAttribute('st-id', namespace);
+        style.setAttribute('st-runtime', runtimeId);
+        style.textContent = css;
+        var loadedStyleElements = head.querySelectorAll<HTMLStyleElement>(
+            `style[st-runtime="${runtimeId}"]`
+        );
+        var inserted = false;
+        for (var i = 0; i < loadedStyleElements.length; i++) {
+            var styleElement = loadedStyleElements[i];
+            var stId = styleElement.getAttribute('st-id');
+            var stDepth = Number(styleElement.getAttribute('st-depth'));
+            if (stId === namespace) {
+                if (stDepth === depth) {
+                    head.replaceChild(style, styleElement);
+                    return;
+                } else {
+                    styleElement.parentElement!.removeChild(styleElement);
+                }
+            }
+            if (!inserted && depth < stDepth) {
+                head.insertBefore(style, styleElement);
+                inserted = true;
```

Comment:
return here? any reason to continue iterating?

Reply:
yes. in order to remove existing nodes with same id